ryanoasis/nerd-fonts
# Autogenerate original-source glyph font

#### Description

**Add original-source.otf generator script**

**[why]**
When we add a custom glyph (or want to update Seti) the process is
rather laborious and we are needed to change the font and the
accompanying `i_seti.sh` in sync.

**[how]**
We use a data file to map icon (svg) filenames to codepoints and
readable names.

That file is parsed and the font and info file is created (overwritten
in the repo); and could then be easily committed. This can be a CI
workflow.

Having a dedicated mapping file (`icons.tsv`) enables us to have stable
codepoints for the same symbol over time. Changes in codepoint
allocation can be checked in git.

Having the font autogenerated help guarantee that the icons are all
likely scaled. We rescale them all to the same size and mid-position.
That is not needed for font-patcher, because it rescales and shifts
again based on to-be-patched font metrics. But it certainly is better
for a view into the original-source font.

Sizes and position are still roughly equivalent to the hand positioned
glyphs.

#### Requirements / Checklist


#### What does this Pull Request (PR) do?

Possibility to generate `src/glyphs/original-source.otf` and `bin/scripts/lib/i_seti.sh` from a file that contains a list of codepoints and all the glyph files in `src/svgs/`.

#### How should this be manually tested?

Make backup of `original-source.otf` and `i_seti.sh`.
Run it (from the script dir). Check changed made to both aforementioned files.

#### Any background context you can provide?

First motivation:
https://github.com/ryanoasis/nerd-fonts/pull/833#issuecomment-1133552076

Came up again and again, for example https://github.com/ryanoasis/nerd-fonts/pull/929#issuecomment-1248497415
when we want to update complete Seti...

#### What are the relevant tickets (if any)?

* #929 
* #922 

#### Screenshots (if appropriate or helpful)

![image](https://user-images.githubusercontent.com/16012374/190865614-04d37fa2-dd2a-4f9b-9886-251d9216e366.png)
